hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -913,11 +913,8 @@ public String startCommit(String actionType, HoodieTableMetaClient metaClient) {
       executeUsingTxnManager(Option.empty(), () -> tryUpgrade(metaClient, Option.empty()));
     }
 
-    CleanerUtils.rollbackFailedWrites(config.getFailedWritesCleanPolicy(),
-        HoodieTimeline.COMMIT_ACTION, () -> tableServiceClient.rollbackFailedWrites());
```

Comment:
why removing this?

Reply:
It is called in startCommitWithTime so we would be calling it twice

Reply:
my bad. I overlooked that you are changing startCommit to startCommitWithTime

Reply:
are we not calling upgrade twice then? 
L912 we can remove it if we are calling into startCommitWithTime atL 917. startCommitWithTime will call upgrade again. So we can avoid the upgrade call in this method